llvm/test/SYCLLowerIR/JointMatrixTransform/constexpr-layout.ll

```diff
@@ -0,0 +1,68 @@
+; The test checks, that unused call to __spirv_AccessChain is eliminated.
+
+; RUN: opt -passes=sycl-joint-matrix-transform < %s -S | FileCheck %s
+
+; ModuleID = 'test.bc'
+source_filename = "test.cpp"
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64-G1"
+target triple = "spir64-unknown-unknown"
+
+$_ZN4sycl3_V16detail26joint_matrix_layout_to_spvENS0_3ext6oneapi12experimental6matrix6layoutE = comdat any
```

Comment:
note: a natural desire would be to remove this function. But unfortunately it's comdat, so it might not be safe to do so.

Reply:
> note: a natural desire would be to remove this function. But unfortunately it's comdat, so it might not be safe to do so.

You need to remove `comdat` from the function definition and then you can safely remove this line as well.

If you pass doesn't have any special handling for `comdats` then you should be fine